src/tensor/index.rs

```diff
@@ -0,0 +1,304 @@
+use crate::Tensor;
```

Comment:
Consider including some module level documentation including a couple examples ?

Reply:
Good idea. I don't have spare time until this weekend. You may merge the PR and I'll write the doc later, or wait for it to happen.